### PR TITLE
Add support for yum/rpm

### PR DESCRIPTION
**Pull request by jonhenrik13**
* rpm/yum support for package_management, software_management and inventory

### CODE REVIEW

File: app/configuration/bundle_software_management.go [hunk 1]
```diff
@@ -178,6 +178,8 @@ func (s Software) Execute(ctx context.Context, srv *Service, pkgManager software
 
 	// restart service if needed
 	if shouldRestart {
+
+		//TODO: We support systemctl and rc.scripts here (OpenWRT, RPM, DEB)
```
Comment (piotrbulinski) [inline app/configuration/bundle_software_management.go:182]: What's the plan for resolving this?

File: app/software/package_manager_deb.go [hunk 2]
```diff
@@ -337,7 +343,7 @@ func (deb *DebianPackageManager) PackageArchitecture() (string, error) {
 }
 
 // ParseDebianPackage and return Package information.
-func ParseDebianPackage(ctx context.Context, pkgFilePath string) (*Package, error) {
+func (deb *DebianPackageManager) ParsePackageFile(ctx context.Context, pkgFilePath string) (*Package, error) {
```
Comment (piotrbulinski) [inline app/software/package_manager_deb.go:346]: Please update comment.

File: app/software/package_manager_rpm.go [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+	"os/exec"
+	"runtime"
+	"strings"
+	"sync"
+
+	"go.qbee.io/agent/app/utils"
+	"go.qbee.io/agent/app/utils/cache"
+)
+
+const PackageManagerTypeRpm PackageManagerType = "rpm"
```
Comment (piotrbulinski) [inline app/software/package_manager_rpm.go:34]: Please add missing comment.

File: app/software/package_manager_rpm.go [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+		}
+	}
+
+	return true
+}
+
+// Busy returns true if package manager is currently busy.
+const yumPidPath = "/var/run/yum.pid"
+
+func (rpm *RpmPackageManager) Busy() (bool, error) {
```
Comment (piotrbulinski) [inline app/software/package_manager_rpm.go:77]: Please move the comment from line 74 here.

File: app/utils/runner/runner.go [hunk 5]
```diff
@@ -12,9 +12,26 @@ import (
 
 // Debian is the image used by the runner.
 const Debian = "debian:qbee"
+const OpenWRT = "openwrt:qbee"
```
Comment (piotrbulinski) [inline app/utils/runner/runner.go:15]: Please add missing comments.

File: app/utils/runner/runner.go [hunk 6]
```diff
@@ -12,9 +12,26 @@ import (
...
+const RHEL = "rhel:qbee"
```
Comment (piotrbulinski) [inline app/utils/runner/runner.go:16]: Please add missing comments.

**Review by piotrbulinski — CHANGES_REQUESTED** [review]
Looks mostly good. Please address the comments situation and let me know what's the plan for the one TODO left.

File: app/configuration/bundle_software_management.go (hunk 1 above)
Comment (jonhenrik13) [reply app/configuration/bundle_software_management.go:182]: This doesn't need to be solved right now, but becomes important when (or if) we do the OpenWRT package manager integration. RHEL based systems use systemd.

Comment (piotrbulinski) [reply app/configuration/bundle_software_management.go]: Should we then just drop the comment? If we need to track any future work, let's do it in Jira.

**Review by piotrbulinski — APPROVED** [review]

**Pull request merged**